tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/AuthorityTests.cs

```diff
@@ -113,6 +115,61 @@ public async Task AuthorityValidationTestWithFalseValidateAuthorityAsync()
                 .ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public async Task UncommonAuthorityValidationTestAsync(bool validateAuthority)
```

Comment:
- Are there existing authority validation and/or instance discovery unit tests that we can just add `WithInstanceDiscovery(true/false)` to, plus extra asserts.
- I think we also need to update tests with disabled instance discovery that cover other scenarios, like user provided metadata, or known-metadata used; to protect from regressions.